Add schema for filter expression

jbeckert opened the pull request:
plus: Correct meta:abstract for offer-representation

jbeckert committed:
Add schema for filter expression
Correct meta:abstract for offer-representation

jbeckert committed:
Add schema for filter expression
Correct meta:abstract for offer-representation

jbeckert committed:
Fix issue with "literal-integer-content" definition

jbeckert committed:
Fix issue with "literal-integer-content" definition

dmitry-shkolnik committed:
Back sync from `xcore`

jbeckert committed:
Add (external) query invocation to filter expression AST

jbeckert committed:
Merge branch 'offer-management' of https://github.com/jbeckert/xdm into offer-management

# Conflicts:
#	extensions/adobe/experience/offer-management/filter-expression.schema.json

jbeckert committed:
Additional examples for the filter expressions.

Outdent the Tags' name into a single string due to requirements adjustment

angusdavisadobe commented on extensions/adobe/experience/offer-management/filter-expression.schema.json:
```diff
@@ -0,0 +1,380 @@
+          "type": "object",
+          "properties": {
+            "nodeType": {
+              "ref": "#/definitions/node-type"
```
should "ref" be "$ref" ?  If not, what is the difference between the two keys?

jbeckert committed:
Fixed PR comments:
use $ref instead of ref in schema references
selection-constraint.eligibility rule is a string not an object
typos in descriptions

jbeckert committed:
Extend filter expression with node type for PQL select expression (generalized map over tuples)
Make sourceProperty string in relationship a proper JSON path expression when the path steps over an array
